test/cbmc/proofs/Sntp_SerializeRequest/Sntp_SerializeRequest_harness.c

```diff
@@ -0,0 +1,45 @@
+/*
+ * coreSNTP v1.0.0
+ * Copyright (C) 2021 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file Sntp_SerializeRequest_harness.c
+ * @brief Implements the proof harness for Sntp_SerializeRequest function.
+ */
+
+#include <stdint.h>
+#include "core_sntp_serializer.h"
+
+void harness()
+{
+    SntpTimestamp_t * pRequestTime;
+    uint32_t randomNumber;
+    void * pBuffer;
+    size_t bufferSize;
+
+    pRequestTime = malloc( sizeof( SntpTimestamp_t ) );
+
+    __CPROVER_assume( bufferSize < CBMC_MAX_OBJECT_SIZE );
+
+    pBuffer = malloc( bufferSize );
+
+    Sntp_SerializeRequest( pRequestTime, randomNumber, pBuffer, bufferSize );
```

Comment:
Is there anything meaningful we can assert about the results of running Sntp_SerializeRequest()?

Reply:
I don't think as it is just serializing the request into the provided buffer and returns the SntpStatus_t.

Reply:
In that case, could `__CPROVER_assert` the return value and contents of `pRequestTime`.

Reply:
To achieve that , a separate function for checking the status value will do as the Sntp_SerializeRequest() can return different status values.